### PR TITLE
Feat: Support multiple modals for use inside for-loops

### PR DESCRIPTION
**Pull request by mapix**
Use multiple streamlit modal in streamlit can cause `key` conflicts,  this pr enhanced isolation.
Thanks for review. 

What's changed:

- Added new Class `Modal` to support modal pop-up by instance
- Added `.gitignore` config for Python 
- Added deprecation warning for old-style singleton modal

### CODE REVIEW

File: streamlit_modal/__init__.py [hunk 1]
```diff
@@ -1,115 +1,160 @@
...
 
 
-def is_open():
-    return st.session_state.get('modal_is_open', False)
+class Modal:
 
+    def __init__(self, title, padding=100, max_width=None, background_color='#fff', key=None):
+        self.title = title
+        self.padding = padding
+        self.backgroud_color = background_color
```
Comment (koenvo) [inline streamlit_modal/__init__.py:14]: Seems `background_color` is not used?

File: setup.py [hunk 2]
```diff
@@ -8,7 +8,8 @@ def setup_package():
 
     setup(
         name="streamlit_modal",
-        version="0.0.5",
+        install_requires=['streamlit', 'deprecation'],
+        version="0.0.6",
```
Comment (koenvo) [inline setup.py:12]: Please change this to 0.1.0.

**Comment by koenvo** [comment]
Hi @mapix, thanks for this pull request! It looks good. Two minor comments.

File: streamlit_modal/__init__.py (hunk 1 above)
Comment (mapix) [reply streamlit_modal/__init__.py]: removed in f75545600d023a331606a13833f6c197e9ded73d

File: setup.py (hunk 2 above)
Comment (mapix) [reply setup.py]: done f75545600d023a331606a13833f6c197e9ded73d

**Comment by mapix** [comment]
I have also changed the pop-up background color from #f0f0f0 to #fff to match streamlit's default style.
it seems better now

图片

**Comment by mapix** [comment]
> Hi @mapix, thanks for this pull request! It looks good. Two minor comments.

all done

**Comment by koenvo** [comment]
> > Hi @mapix, thanks for this pull request! It looks good. Two minor comments.
> 
> all done

Oh, sorry I missed the line where `self.background_color` was used. I think it's a nice feature. I didn't mean to remove it, only point was that it looked like a non-used variable

**Comment by mapix** [comment]
@koenvo 
may rare needed, left `#fff` seems good enough :-)

**Comment by sfc-gh-pkommini** [comment]
image
Backgroud color is needed for streamlit apps where the theme is set to dark mode. The pop-up color still uses white.

**Pull request merged**